platform/darwin/src/MGLConversionValue.h

```diff
@@ -0,0 +1,73 @@
+//
+//  MGLConversionValue.h
+//  ios
+//
+//  Created by Anand Thakker on 2/10/17.
+//  Copyright © 2017 Mapbox. All rights reserved.
+//
+
+#ifndef MGLConversionValue_h
+#define MGLConversionValue_h
+
+class MGLConversionValue {
+public:
+    Value(NSObject * _value) : value(_value) {}
+    
+    bool isNull() const {
+        return value;
+    }
+    
+    bool isArray() const {
+        return [value isKindOfClass:[NSArray class]];
+    }
+    
+    bool isObject() const {
+        return [value isKindOfClass:[NSDictionary class]];
+    }
+    
+    bool isString() const {
+        return [value isKindOfClass:[NSString class]];
+    }
+    
+    bool isBool() const {
+        return [value isKindOfClass:[NSNumber class]] &&
+                ((strcmp([value objCType], @encode(char)) == 0) ||
+                (strcmp([value objCType], @encode(BOOL)) == 0));
+    }
+    
+    bool isNumber() const {
+        return [value isKindOfClass:[NSNumber class]] && !isBool();
+    }
+    
+    std::string toString() const {
+        return std::string([(NSString *)value UTF8String]);
+    };
+    
+    float toNumber() const {
```

Comment:
> By the way, is replacing mgl_constantMBGLValue something I should consider as a requirement/use case for MGLConversionValue?

Not necessarily, although we should make sure they're somewhat consistent. mgl_constantMBGLValue is primarily intended as a way to convert constant NSExpressions representing vector tile feature properties into the equivalent mbgl types. This is why it makes a distinction between integers and floats, for instance, and doesn't handle enumeration values. This method is used when converting an NSPredicate (which is made up of NSExpressions) into an mbgl::Filter (which generally contains mbgl::Values).